### ISSUE TITLE
No error on double root

### ISSUE THREAD

**Issue by jazzzz** [isaacs/sax-js#168]
This XML should cause an error, currently it does not.

```xml
<root></root>
<root2></root2>
```

**Comment by george-hawkins-aa** [comment]
Yes - this was the first thing I tried when I was writing a few tests to see whether sax-js would choke in a meaningful way on encountering various problems.

I've seen the comments in #154 about root elements and the XML spec. And I know that it says in the main README that sax-js doesn't try too hard on the validation front.

But might it be at least possible to check for this situation. I have non-expert users hand generating XML files (yes - I know this probably indicates something badly wrong with the process involved) and I'd like to catch as much obvious stuff before issues get picked up by stricter validating downstream systems.

**Comment by ajvincent** [comment]
https://github.com/isaacs/sax-js/pull/182
